File: e2e/playwright-typescript/interactions/validators/LanguageValidator.ts
import { expect, Page } from '@playwright/test';
import { MainPage } from '@/interactions/pages/MainPage';
import { step } from '@/decorators/step-decorator';

export class LanguageValidator {
  private readonly mainPage: MainPage;

  constructor(page: Page) {
    this.mainPage = new MainPage(page);
  }

  @step('Validate UI is in Spanish')
  async uiIsInSpanish(): Promise<void> {
    await expect(this.mainPage.createTaskButton).toHaveText('Crear tarea');
    await expect(this.mainPage.tableHeaders).toContainText('Título');
    await expect(this.mainPage.tableHeaders).toContainText('Estado');
    await expect(this.mainPage.tableHeaders).toContainText('Prioridad');
  }

  @step('Validate status tag for {status} shows {expectedText}')
  async statusTagShowsText(status: string, expectedText: string): Promise<void> {
    await expect(this.mainPage.statusTag(status)).toHaveText(expectedText);
  }

  @step('Validate priority tag for {priority} shows {expectedText}')
  async priorityTagShowsText(priority: string, expectedText: string): Promise<void> {
    await expect(this.mainPage.priorityTag(priority)).toHaveText(expectedText);
  }
}

File: e2e/playwright-python/interactions/validators/language_validator.py
from playwright.sync_api import Page, expect
from interactions.pages.main_page import MainPage
from decorators.step_decorator import step


class LanguageValidator:
    def __init__(self, page: Page):
        self._main_page = MainPage(page)

    @step("Validate UI is in Spanish")
    def ui_is_in_spanish(self) -> None:
        expect(self._main_page.create_task_button).to_have_text("Crear tarea")
        expect(self._main_page.table_headers).to_contain_text("Título")
        expect(self._main_page.table_headers).to_contain_text("Estado")
        expect(self._main_page.table_headers).to_contain_text("Prioridad")

    @step("Validate status tag for {status} shows {expected_text}")
    def status_tag_shows_text(self, status: str, expected_text: str) -> None:
        expect(self._main_page.status_tag(status)).to_have_text(expected_text)

    @step("Validate priority tag for {priority} shows {expected_text}")
    def priority_tag_shows_text(self, priority: str, expected_text: str) -> None:
        expect(self._main_page.priority_tag(priority)).to_have_text(expected_text)
